lib/ansible/modules/windows/win_psrepository.py

```diff
@@ -0,0 +1,73 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+
+# Copyright: (c) 2018, Wojciech Sciesinski <wojciech[at]sciesinski[dot]net>
+# Copyright: (c) 2017, Daniele Lazzari <lazzari@mailup.com>
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+# this is a windows documentation stub.  actual code lives in the .ps1
+# file of the same name
+
+ANSIBLE_METADATA = {'metadata_version': '1.1',
+                    'status': ['preview'],
+                    'supported_by': 'community'}
+
+DOCUMENTATION = '''
+---
+module: win_psrepository
+version_added: "2.8"
+short_description: Adds, removes or updates a Windows PowerShell repository.
+description:
+  - This module helps to install, remove and update Windows PowerShell repository on Windows-based systems.
+options:
+  name:
+    description:
+      - Name of the repository to work with.
+    aliases:
+      - repository
+    required: yes
+  source_location:
+    description:
+      - Specifies the URI for discovering and installing modules from this repository.
```

Comment:
I tried to find a better explanation but I failed. Based at [PowerShell\PowerShellGet](https://github.com/PowerShell/PowerShellGet), specially the code of the function [Get-LocationString](https://github.com/PowerShell/PowerShellGet/blob/development/src/PowerShellGet/private/functions/Get-LocationString.ps1) it just `[uri]`. The Uri .Net class [description](https://docs.microsoft.com/en-us/dotnet/api/system.uri?view=netframework-4.7.2) can help but I'll analyze that deeper at weekend and maybe I'll open an issue for the PowerShellGet module documentation.

So, for now, please accept the explanation exact like for the command `Register-PSRepository`

Reply:
What I'm trying to say is that it does work and I've used it before, use this script as an example

```
$repo_name = "local repo"
$path = "C:\repos"
$url = "https://devopsgallerystorage.blob.core.windows.net/packages/powershell-yaml.0.3.7.nupkg"
if (Test-Path -Path $path) {
    Remove-Item -Path $path -Force -Recurse
}
New-Item -Path $path -ItemType Directory > $null
Invoke-WebRequest -Uri $url -OutFile "$path\powershell-yaml.0.3.7.nupkg"

if (Get-PSRepository -Name $repo_name -ErrorAction SilentlyContinue) {
    Unregister-PSRepository -Name $repo_name
}
Register-PSRepository -Name $repo_name -SourceLocation $path
Find-Module -Name powershell-yaml
```

The output I get is

```
PS C:\Windows\system32> Find-Module -Name powershell-yaml

Version              Name                                Repository           Description
-------              ----                                ----------           -----------
0.3.7                powershell-yaml                     PSGallery            Powershell module for serializing and ...
0.3.7                powershell-yaml                     local repo           Powershell module for serializing and ...
```
Which tells me it is also using the local filesystem as a repository location. There are a few blog posts that also detail this like https://kevinmarquette.github.io/2017-05-30-Powershell-your-first-PSScript-repository/ and https://4sysops.com/archives/how-to-create-a-file-share-powershell-repository/.

Lastly while the `-SourceLocation` type is a URI, a URI is not just a URL. It can represent different resource paths, like a URL, FTP, file and so on. The full URI for a local path is `file:///C:/path/to/folder` and .NET will automatically convert a path to the full URL path, e.g.

```
PS C:\Windows\system32> New-Object -TypeName System.Uri -ArgumentList "C:\temp"


AbsolutePath   : C:/temp
AbsoluteUri    : file:///C:/temp
LocalPath      : C:\temp
Authority      :
HostNameType   : Basic
IsDefaultPort  : True
IsFile         : True
IsLoopback     : True
PathAndQuery   : C:/temp
Segments       : {/, C:/, temp}
IsUnc          : False
Host           :
Port           : -1
Query          :
Fragment       :
Scheme         : file
OriginalString : C:\temp
DnsSafeHost    :
IdnHost        :
IsAbsoluteUri  : True
UserEscaped    : False
UserInfo       :
```

Reply:
I've added an additional explanation.